web/ui/react-app/src/App.tsx

```diff
@@ -5,33 +5,61 @@ import { Container } from 'reactstrap';
 import './App.css';
 import { Router, Redirect } from '@reach/router';
 import { Alerts, Config, Flags, Rules, ServiceDiscovery, Status, Targets, TSDBStatus, PanelList } from './pages';
-import PathPrefixProps from './types/PathPrefixProps';
 
 interface AppProps {
   consolesLink: string | null;
 }
 
-const App: FC<PathPrefixProps & AppProps> = ({ pathPrefix, consolesLink }) => {
+const App: FC<AppProps> = ({ consolesLink }) => {
+  // this dynamically/generically determines the pathPrefix by taking the window location path
+  // and stripping the known endpoint suffixes. works out of the box for both direct hosting
+  // and reverse proxy deployments with no additional configurations required
+  let basePath = window.location.pathname;
+  const paths = [
+    '/graph',
+    '/alerts',
+    '/status',
+    '/tsdb-status',
+    '/flags',
+    '/config',
+    '/rules',
+    '/targets',
+    '/service-discovery',
+    '/',
```

Comment:
Just for my understanding, are you including this `/` path just for completeness' sake, although it will never be matched below because we only check when the base path length is > 1?